### PR TITLE
Questions and answers - added functionality to the components

### PR DESCRIPTION
**Pull request by Katerina-spb**
This is what I got so far.

Still need to work on 
search component,
filtering the questions,
updating the component when new question or answer submitted

### CODE REVIEW

**Comment by Katerina-spb** [comment]
sorry my commits are not that clean...

File: src/apiController.js [hunk 1]
```diff
@@ -113,7 +112,7 @@ let apiController = {
     let params = { page: pages, count: count };
 
     return axios
-      .get(`/qa/questions/${questionId}/{answers`, { params })
```
Comment (chaines) [inline src/apiController.js:116]: Obviously this small change here is fine, but if you're going to have to make any large changes to the apiController in the future, make sure those changes are commited seperately (and ASAP), to reduce merge conflicts as much as possible.

File: src/components/questions/Questions.css [hunk 2]
```diff
@@ -4,21 +4,11 @@
...
   display: flex;
   flex-direction: column;
 
 }
 
-.q_list_btns {
-  display: flex;
-  margin-left: 15px;
-  margin-top: 20px;
-}
```
Comment (chaines) [inline src/components/questions/Questions.css:24]: I just want to mention here that I have a really bad habit of not deleting css I'm not using any more, so I'm glad to see you don't have that problem :)

File: src/components/questions/Questions.js [hunk 3]
```diff
@@ -5,18 +5,12 @@ import './Questions.css';
 
 var Questions = () => {
   return (
-    <div class = 'questions_widget_container'>
-      <div class = 'search_form'>
+    <div className = 'questions_widget_container'>
+      <div className = 'search_form'>
       <Search />
       </div>
```
Comment (chaines) [inline src/components/questions/Questions.js]: Just a thought here:  if the only thing inside the `search_form` div is the `Search` component, you might be better served putting that div inside the search component itself, though there are certainly arguments that could be made for leaving it here.

File: src/components/questions/answer/Answer.js [hunk 4]
```diff
@@ -1,19 +1,60 @@
...
+    'July',
+    'August',
+    'September',
+    'October',
+    'November',
+    'December'
+  ];
+  var monthName = months[date.getMonth()];
+  var formatDate = `${monthName} ${day}, ${year}`;
+
```
Comment (chaines) [inline src/components/questions/answer/Answer.js:14]: Just wanted to point out that all of the date code here from line 13-30 could be replaced with
 `var formatDate = date.toLocaleDateString(undefined, { year: 'numeric', month: 'long', day: 'numeric' })`
 which has the added benefit of putting months in the language of the user, as well as localizing the order of the date string (so, in most european formats it would be "Day month year" rather than "Month day, year".
 
 The code is fine and works as intended, just something I would do to reduce clutter

File: src/components/questions/answerList/AnswerList.css [hunk 5]
```diff
@@ -1,5 +1,19 @@
 .load_answers {
```
Comment (chaines) [inline src/components/questions/answerList/AnswerList.css]: Once again, just a style nitpick, but `load_answers` isn't a very descriptive class name, especially when it's being used on a button that *doesn't* load more answers.

File: src/components/questions/helpfulness/Helpfulness.js [hunk 6]
```diff
@@ -1,14 +1,38 @@
...
-</div>
+    <div className = 'help'>
+      Helpful?
+      <a href ='#'
+      onClick = {() => handleClick(id)}
+      className = 'add_link'>
+      Yes
+      </a>
+      ({helpfulness}) |
+    </div>
```
Comment (chaines) [inline src/components/questions/helpfulness/Helpfulness.js:34]: Something to consider here, from a user experience perspective, is only have "Yes" be a link if they haven't already rated it. It would require the use of another ternary operator, but I think it would be nice.

Something like this maybe:
```
<div className = 'help'>
  Helpful?
  {rated ? '✓' : <a href="#" onClick = {() => handleClick(id)} className = 'add_link'>Yes</a>}
  ({helpfulness})
</div>
```
(note, I haven't tested the above, I just wrote it in this comment, but I would assume it works.)

File: src/components/questions/answerList/AnswerList.js [hunk 7]
```diff
@@ -1,18 +1,68 @@
...
+    getAnswers(questionId, {params})
+  }, []);
+
+  useEffect(() => {
+    getAnswers(questionId, {params})
+  }, [questionId]);
+
+
+  if (answers.length === 0) {
+    return <em style = {{color: 'gray'}}>There is no answers for this question yet</em>;
```
Comment (chaines) [inline src/components/questions/answerList/AnswerList.js:33]: Grammar check: "There *are* no answers for this question yet"

File: src/components/questions/helpfulness/Helpfulness.css [hunk 8]
```diff
@@ -1,3 +1,7 @@
 .help {
   font-size: small;
+}
+
+.add_link {
+  cursor: default;
```
Comment (chaines) [inline src/components/questions/helpfulness/Helpfulness.css:6]: Out of curiosity, why did you add this? Was it a part of the business requirements? Generally you usually want something that is clickable to have the 'pointer' cursor

File: src/components/questions/modals/AnswerForm.js [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+      photos: []
+    }
+    apiController.postAnswer(questionId, params);
+  }
+
+  return (
+    <form onSubmit = {handleSubmit} autoComplete="off">
+      <div className = 'form-container'>
+       <TextField
+        required id="standard-required"
```
Comment (chaines) [inline src/components/questions/modals/AnswerForm.js]: I can't find any documentation on this `standard-required` id. Do you know where you found it/why this one is `standard-required` while the others are `standard-basic`?

File: src/components/questions/modals/AnswerForm.js [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+      /> <br />
+
+      <TextField
+        required id="standard-basic"
+        label="What is your email"
+        placeholder = 'Example: jack@email.com'
+        fullWidth value={email}
+        onChange={e => setEmail(e.target.value)}
+        helperText = 'For authentication reasons, you will not be emailed'
+      />
```
Comment (chaines) [inline src/components/questions/modals/AnswerForm.js:57]: Another style nitpick: This file would be a lot easier to read if the props on all your text fields were in the same order. I would also put the 'boolean' props (required and fullWidth) on a line together, rather than putting them on a line with other props. Because reading through the code, it looks like required and id are related props, and fullWidth and value are related props.

File: src/components/questions/modals/AnswerModal.js [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+        display: 'flex',
+        alignItems: 'center',
+        justifyContent: 'center',
+
+    },
+    paper: {
+        backgroundColor: theme.palette.background.paper,
+        border: '2px solid #000',
+        boxShadow: theme.shadows[5],
+        padding: theme.spacing(4, 10, 6),
```
Comment (chaines) [inline src/components/questions/modals/AnswerModal.js:19]: This use of theme isn't something I came across when looking at `makeStyles` docs, I'm going to have to look into it and incorporate it into my styles.

File: src/components/questions/modals/AnswerModal.js [hunk 12]
```diff
@@ -0,0 +1,80 @@
...
+        setOpen(false);
+    };
+
+    const mystyle = {
+      backgroundColor: "white",
+      padding: "0px",
+      fontSize: "small",
+      border: 'none',
+      cursor: 'pointer',
+    }
```
Comment (chaines) [inline src/components/questions/modals/AnswerModal.js]: You should go ahead and move this into the makeStyles function above, so that your styles are all in one place in the file

File: src/components/questions/modals/AnswerModal.js [hunk 13]
```diff
@@ -0,0 +1,80 @@
...
+
+    return (
+        <div>
+
+            <button style = {mystyle} onClick={handleOpen}>
+                Add Answer
+            </button>
```
Comment (chaines) [inline src/components/questions/modals/AnswerModal.js]: is there a specific reason you're using `<button>` here rather than the Material-UI `<Button>` component?

File: src/components/questions/questionList/QuestionList.css [hunk 14]
```diff
@@ -0,0 +1,9 @@
+.q_list_btns {
+  display: flex;
+  margin-left: 15px;
+}
+
+button {
+  padding: 5px;
+  margin-right: 20px;
+}
```
Comment (chaines) [inline src/components/questions/questionList/QuestionList.css:9]: be careful with very broad selectors like `button`, as they'll impact everything on the page, including buttons in other components/widgets.

**Review by chaines — CHANGES_REQUESTED** [review]
The only 'broad' comment I have is you are pretty inconsistent with your usage of `var` vs `let` vs `const`. You should probably not being using `var` at all, and rather `let` if it's a variable that you may need to change or `const` otherwise. 

There's a lot of comments on this PR, but most of it is just stuff to think about. The only thing I'm uncomfortable merging into master right now is the `button` selector in `QuestionList.css`. Get rid of that global selector and we can go ahead and get all this code into master as is.

File: src/apiController.js (hunk 1 above)
Comment (Katerina-spb) [reply src/apiController.js:116]: my bad!

File: src/components/questions/answer/Answer.js (hunk 4 above)
Comment (Katerina-spb) [reply src/components/questions/answer/Answer.js:14]: thanks!

File: src/components/questions/modals/AnswerModal.js (hunk 11 above)
Comment (Katerina-spb) [reply src/components/questions/modals/AnswerModal.js:19]: Up to you! I can delete it if we don't want to use it

Comment (chaines) [reply src/components/questions/modals/AnswerModal.js:19]: No, it's completely fine! I just didn't even realize the functionality existed!

**Review by chaines — APPROVED** [review]
There were a couple comments that were being hidden by github by default that are still worth taking a look at in the future, but I'm happy with this as is, and will merge into master. Good job Kate!
